app/models/question.rb

```diff
@@ -69,7 +69,7 @@ class Question < ApplicationRecord
            :odk_tag,
            :odk_name,
            :select_multiple?,
-           to: :qtype
+           to: :qtype, allow_nil: true
```

Comment:
This approach works but it has the unfortunate side effect of changing the method contract of 9 different methods on this class—specifically it means that they all may return nil, where that wasn't possible before. This is a big problem because it may violate the assumptions of all kinds of code throughout the app. Code that didn't need to check for a nil value when calling e.g. `printable?` now does, or should, or otherwise it is vulnerable to hard-to-debug errors. This is why nils need to be used quite carefully and are actually considered a code smell in some languages!

I think a better way to fix this is to change this line in `Questioning`:

```
  validates_with Forms::DynamicPatternValidator, field_name: :default, force_calc_if: :numeric?
```

Instead of having it provide the symbol `:numeric?` for `force_calc_if`, have it provide a Proc and then in that proc you can do e.g.  `->(qing) { qing.qtype&.numeric? }`. Then you will have to update the DynamicPatternValidator to call the proc instead of using `send` to call the named method. Since Questioning is the only class using `force_calc_if`, it's fine to swap out this behavior.

Or if you have another suggestion of a way to avoid changing the contract for all these methods, feel free to propose it!

Do you see the difference and why it's important?